tiatoolbox/models/engine/nucleus_detector.py

```diff
@@ -0,0 +1,1126 @@
+                Expected prediction dtype.
+            **kwargs (NucleusDetectorRunParams):
+                Additional runtime parameters to configure segmentation.
+
+                Optional Keys:
+                    min_distance (int):
+                        Minimum distance separating two nuclei (in pixels).
+                    threshold_abs (float):
+                        Absolute detection threshold applied to model outputs.
+                    threshold_rel (float):
+                        Relative detection threshold
+                        (e.g., with respect to local maxima).
+                    postproc_tile_shape (tuple[int, int]):
+                        Tile shape (height, width) for post-processing rechunking.
+                    cache_dir (str or os.PathLike):
+                        Directory for caching intermediate centroid maps as Zarr.
+                        Defaults to './tmp/'.
+
+        Returns:
+            dict[str, da.Array]:
+                A dictionary mapping detection fields to 1-D Dask arrays:
+                - ``"x"``: x coordinates of detected nuclei (``np.uint32``).
+                - ``"y"``: y coordinates of detected nuclei (``np.uint32``).
+                - ``"classes"``: class IDs (``np.uint32``).
+                - ``"probabilities"``: detection scores (``np.float32``).
+
+        Notes:
+            - Halo padding ensures that nuclei crossing tile/chunk boundaries
+              are not fragmented or duplicated.
+            - If thresholds are not explicitly provided, model defaults are used.
+            - Centroid maps are computed and saved to Zarr storage to avoid
+              out-of-memory errors on large WSIs.
+            - The Zarr-backed centroid maps are then processed block-by-block
+              to extract detections incrementally.
+
+        """
+        _ = prediction_shape
+
+        logger.info("Post processing WSI predictions in NucleusDetector")
+
+        # If these are not provided, defaults from model will be used in postproc
+        threshold_abs = kwargs.get("threshold_abs")
+        threshold_rel = kwargs.get("threshold_rel")
+
+        # min_distance and postproc_tile_shape cannot be None here
+        min_distance = kwargs.get("min_distance")
+        if min_distance is None:
+            min_distance = self.model.min_distance
+        postproc_tile_shape = kwargs.get("postproc_tile_shape")
+        if postproc_tile_shape is None:
+            postproc_tile_shape = self.model.postproc_tile_shape
+
+        # Add halo (overlap) around each block for post-processing
+        depth_h = min_distance
+        depth_w = min_distance
+        depth = {0: depth_h, 1: depth_w, 2: 0}
+
+        # Re-chunk to post-processing tile shape for more efficient processing
+        rechunked_prediction_map = raw_predictions.rechunk(
+            (postproc_tile_shape[0], postproc_tile_shape[1], -1)
+        )
+
+        centroid_maps = da.map_overlap(
```

Comment:
Swap the order of the first two args to avoid this depreciation:

FutureWarning: The use of map_overlap(array, func, **kwargs) is deprecated since dask 2.17.0 and will be an error in a future release.

Reply:
Please can you share the source for this warning?